src/libmusig2.c

```diff
@@ -385,18 +387,16 @@ int musig2_aggregate_partial_sig(musig2_partial_signature *mps, unsigned char *s
     /* Aggregate the partial signatures */
     memcpy(signature + MUSIG2_AGGR_PUBKEY_BYTES, mps[0].sig, MUSIG2_SCALAR_BYTES);
     for (i = 1; i < nr_signatures; i++) {
-        if (!secp256k1_ec_seckey_tweak_add(ctx, signature + MUSIG2_AGGR_PUBKEY_BYTES, mps[i].sig)) {
-            secp256k1_context_destroy(ctx);
-            return 0;
+        if (!secp256k1_ec_seckey_tweak_add(ctx, signature + MUSIG2_AGGR_PUBKEY_BYTES, mps[i].sig)){
+            return MUSIG2_ERR_ADD_PARSIG;
         }
-
     }
-    secp256k1_context_destroy(ctx);
+secp256k1_context_destroy(ctx);
 
-    return 1;
+    return MUSIG2_OK;
 }
 
-void musig2_prepare_verifier(musig2_aggr_pubkey *aggr_pk, unsigned char *serialized_pk_list, int nr_signers) {
+MUSIG2_ERROR musig2_prepare_verifier(secp256k1_xonly_pubkey *aggr_pk, unsigned char *serialized_pk_list, int nr_signers) {
```

Comment:
The type of aggr_pk should not be changed. I can see that in `.h` file you did not change it. Maybe a leftover?